lib/ohai/plugins/linux/interrupts.rb

```diff
@@ -0,0 +1,86 @@
+#
+# Author:: Davide Cavalca <dcavalca@fb.com>
+# Copyright:: Copyright (c) 2020 Facebook
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Ohai.plugin(:Interrupts) do
+  depends "cpu"
+  provides "interrupts", "interrupts/irq", "interrupts/smp_affinity_by_cpu"
+
+  # Documentation: https://www.kernel.org/doc/Documentation/IRQ-affinity.txt
+  # format: comma-separate list of 32bit bitmask in hex
+  # each bit is a CPU, right to left ordering (i.e. CPU0 is rightmost)
+  def parse_smp_affinity(path, cpus)
+    masks = File.read(path).strip
+    bit_masks = []
+    masks.split(",").each do |mask|
+      bit_masks << mask.rjust(8, "0").to_i(16).to_s(2)
+    end
+    affinity_mask = bit_masks.join
+    affinity_by_cpu = affinity_mask.split("").reverse
+    smp_affinity_by_cpu = Mash.new
+    (0..cpus - 1).each do |cpu|
+      smp_affinity_by_cpu[cpu] = affinity_by_cpu[cpu].to_i == 1
+    end
+    smp_affinity_by_cpu
+  end
+
+  collect_data(:linux) do
+    interrupts Mash.new
+
+    cpus = cpu["total"]
+    interrupts[:smp_affinity_by_cpu] =
+      parse_smp_affinity("/proc/irq/default_smp_affinity", cpus)
+
+    interrupts[:irq] = Mash.new
+    File.open("/proc/interrupts").each do |line|
+      # Documentation: https://www.kernel.org/doc/Documentation/filesystems/proc.txt
+      # format is "{irqn}: {CPUn...} [type] [vector] [device]"
+      irqn, fields = line.split(":", 2)
+      # skip the header
+      next if fields.nil?
+
+      irqn.strip!
+      Ohai::Log.debug("irq: processing #{irqn}")
+
+      interrupts[:irq][irqn] = Mash.new
+      interrupts[:irq][irqn][:events_by_cpu] = Mash.new
+
+      fields = fields.split(" ", cpus + 1)
+      (0..cpus - 1).each do |cpu|
+        if /\d+/.match(fields[cpu])
```

Comment:
did you want ^ and $ on this?
Also the less-expensive way to do this is `!!Integer(fields[cpu])`

Reply:
`!!Integer(fields[cpu])` doesn't work here because this could be `nil`.

Reply:
I don't think anchoring the regexp makes any difference here, as we're just matching on the field, not the whole line.

Reply:
Sure but you'll match this if it's got an integer ANYWHERE in it, not if it *is* an integer. And yes `!!Integer(fields[cpu])` will work You'd use it like this:

````
cpuval = nil
begin
  !!Integer(fields[cpu])
rescue ArgumentError
  pass
end
if cpuval
....
```

BUt as I said a regex is also fine, if you anchor it. But as it stands if you get "p1xel" as a value, you'll try to `.to_i` which probably isn't waht you want.

Although, in fairness, `to_i` turns all things into 0 it doesn't understand, so maybe you do'nt want to test it at all and JUST want to use `to_i`

Reply:
Yep, you're right, `to_i` just does the right thing here, so I'll use that instead. I've also anchored the regexp below.